libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1472,13 +1484,57 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context.appendConditionalInvalid();
 
 		if (_operator == Token::Div)
-			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
+		{
+			if (fixedType)
+			{
+				// Division would cancel out factor, this gives the numerator an extra factor so it remains in result
+				m_context << u256(pow(bigint(10), fixedType->fractionalDigits()))
+						  << Instruction::MUL;
+			}
+			m_context << (isSigned ? Instruction::SDIV : Instruction::DIV);
+		}
 		else
-			m_context << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+			m_context << (isSigned ? Instruction::SMOD : Instruction::MOD);
 		break;
 	}
 	case Token::Exp:
-		m_context << Instruction::EXP;
+		if (intType)
+		{
+			m_context << Instruction::EXP;
+		}
+		else
+		{
+			std::string code = R"(
+			{
+				if lt(exponent, 0) {
+					invalid()
```

Comment:
I would actually prefer to disallow that as part of the type system: `a**b` is invalid if `b` is a signed fractional type.